### PR TITLE
Add "week of year" expression

### PR DESCRIPTION
**Pull request by isovector**
My current contract for work is to do X hours of work a week. I use arbtt for everything else, and would like to have a convenient means of checking how much work I've done this calendar week.

This PR adds support for it.

### CODE REVIEW

**Review by nomeata — APPROVED** [review]
Thanks! Nice to get contributions on kinda old projects :-)

Looks good to me. I assume you did actually test and use this code?

**Comment by isovector** [comment]
I've added the following to my config:

```
week of year $now == week of year $date ==> tag this-week,
```

and it is working as expected :)

File: src/Categorize.hs [hunk 1]
```diff
@@ -354,6 +355,15 @@ evalDayOfMonth cp = Left $ printf
   "Cannot apply day of month to an expression of type %s, only to $date."
   (cpType cp)
 
+-- Day of month is an integer in [1..31].
```
Comment (isovector) [inline src/Categorize.hs:358]: Oops, stale comment

**Comment by isovector** [comment]
I've updated the documentation with the new expression!

**Pull request merged**